src/Native/Runtime/arm/ExceptionHandling.S

```diff
@@ -90,95 +167,315 @@ NESTED_END RhpThrowEx, _TEXT
+        movt        r2, #:upper16:TSF_DoNotTriggerGc
+        bics        r1, r2
+        strex       r2, r1, [r0, #OFFSETOF__Thread__m_ThreadStateFlags]
+        cbz         r2, LOCAL_LABEL(ClearSuccess)
+        b           LOCAL_LABEL(ClearRetry)
+LOCAL_LABEL(ClearSuccess):
+
+        mov         r2, r4        // reload handler funclet address
+        mov         r1, r5        // reload REGDISPLAY pointer
+
+        //
+        // set preserved regs to the values expected by the funclet
+        //
+        ldr         r12, [r1, #OFFSETOF__REGDISPLAY__pR4]
+        ldr         r4, [r12]
+        ldr         r12, [r1, #OFFSETOF__REGDISPLAY__pR5]
+        ldr         r5, [r12]
+        ldr         r12, [r1, #OFFSETOF__REGDISPLAY__pR6]
+        ldr         r6, [r12]
+        ldr         r12, [r1, #OFFSETOF__REGDISPLAY__pR7]
+        ldr         r7, [r12]
+        ldr         r12, [r1, #OFFSETOF__REGDISPLAY__pR8]
+        ldr         r8, [r12]
+        ldr         r12, [r1, #OFFSETOF__REGDISPLAY__pR9]
+        ldr         r9, [r12]
+        ldr         r12, [r1, #OFFSETOF__REGDISPLAY__pR10]
+        ldr         r10, [r12]
+        ldr         r12, [r1, #OFFSETOF__REGDISPLAY__pR11]
+        ldr         r11, [r12]
+
+        //
+        // load vfp preserved regs
+        //
+        add         r12, r1, #OFFSETOF__REGDISPLAY__D
+        vldm        r12!, {d8-d15}
+
+        ldr         r0, [r1, #OFFSETOF__REGDISPLAY__SP] // r0 <- establisher frame
+        blx         r2                                  // handler funclet address
 
         EXPORT_POINTER_TO_ADDRESS PointerToRhpCallFinallyFunclet2
 
+        ldr         r1, [sp, #rsp_offset_r1]        // reload REGDISPLAY pointer
+
+        //
+        // save new values of preserved regs into REGDISPLAY
+        //
+        ldr         r12, [r1, #OFFSETOF__REGDISPLAY__pR4]
+        str         r4, [r12]
+        ldr         r12, [r1, #OFFSETOF__REGDISPLAY__pR5]
+        str         r5, [r12]
+        ldr         r12, [r1, #OFFSETOF__REGDISPLAY__pR6]
+        str         r6, [r12]
+        ldr         r12, [r1, #OFFSETOF__REGDISPLAY__pR7]
+        str         r7, [r12]
+        ldr         r12, [r1, #OFFSETOF__REGDISPLAY__pR8]
+        str         r8, [r12]
+        ldr         r12, [r1, #OFFSETOF__REGDISPLAY__pR9]
+        str         r9, [r12]
+        ldr         r12, [r1, #OFFSETOF__REGDISPLAY__pR10]
+        str         r10, [r12]
+        ldr         r12, [r1, #OFFSETOF__REGDISPLAY__pR11]
+        str         r11, [r12]
+
+        //
+        // store vfp preserved regs
+        //
+        add         r12, r1, #OFFSETOF__REGDISPLAY__D
+        vstm        r12!, {d8-d15}
+
+        //
+        // set the DoNotTriggerGc flag, trashes r0-r2
+        //
+        INLINE_GETTHREAD      // r0 <- Thread*
+LOCAL_LABEL(SetRetry):
+        ldrex       r1, [r0, #OFFSETOF__Thread__m_ThreadStateFlags]
+        movw        r2, #:lower16:TSF_DoNotTriggerGc
```

Comment:
@jkotas I did not define value of TSF_DoNotTriggerGc before and so got the message that it's arm instruction. Fixed. Thanks